app/assets/javascripts/discourse/app/components/sidebar/messages-section.js

```diff
@@ -1,10 +1,99 @@
 import { action } from "@ember/object";
+import { cached } from "@glimmer/tracking";
 
 import GlimmerComponent from "discourse/components/glimmer";
 import Composer from "discourse/models/composer";
 import { getOwner } from "discourse-common/lib/get-owner";
+import GroupMessageSectionLink from "discourse/lib/sidebar/messages-section/group-message-section-link";
+import PersonalMessageSectionLink from "discourse/lib/sidebar/messages-section/personal-message-section-link";
+
+export const INBOX = "inbox";
+const UNREAD = "unread";
+const SENT = "sent";
+const NEW = "new";
+const ARCHIVE = "archive";
+
+export const PERSONAL_MESSAGES_INBOXES = [INBOX, UNREAD, NEW, SENT, ARCHIVE];
+export const GROUP_MESSAGES_INBOXES = [INBOX, UNREAD, NEW, ARCHIVE];
 
 export default class SidebarMessagesSection extends GlimmerComponent {
+  constructor() {
+    super(...arguments);
+
+    this.appEvents.on(
+      "page:changed",
```

Comment:
Isn't this the same event that backs the `onPageChange` method in our plugin API? If so, I remember that relying on this event to update the UI isn't ideal because it fires too late in the route transition lifecycle. Is it possible to use something else here? Maybe the [router service](https://api.emberjs.com/ember/release/classes/RouterService) can be used here? It has a `currentRouteName` property. If it's not possible or too difficult, then no problem, we can live with it for now and revisit it later.

Reply:
>  it fires too late in the route transition lifecycle

Actually I need it to fire at the end of the transition lifecycle. Initially I relied on the router service but `currentRouteName` will be in multiple states while loading a page. For example, there is `userPrivateMessages.loading`, `user.loading` etc...  Handling those states made the code more complicated to reason about.

Reply:
Aha got it. It's rare to find an example where it makes sense to use the `page:changed` event to update the UI, but this is a good example :ok_hand: